examples/comp_w_dac.rs

```diff
@@ -0,0 +1,63 @@
+// #![deny(warnings)]
+#![deny(unsafe_code)]
+#![no_main]
+#![no_std]
+
+use embedded_hal::Direction;
+use hal::comparator::{self, ComparatorExt, ComparatorSplit};
+use hal::dac::{Dac1IntSig1, DacExt, DacOut};
+use hal::delay::SYSTDelayExt;
+use hal::gpio::GpioExt;
+use hal::rcc::RccExt;
+use stm32g4xx_hal as hal;
+mod utils;
+extern crate cortex_m_rt as rt;
+
+use hal::stm32;
+use rt::entry;
+
+#[entry]
+fn main() -> ! {
+    let dp = stm32::Peripherals::take().expect("cannot take peripherals");
+    let cp = cortex_m::Peripherals::take().expect("cannot take core peripherals");
+
+    let mut rcc = dp.RCC.constrain();
+    let mut delay = cp.SYST.delay(&rcc.clocks);
+
+    let gpioa = dp.GPIOA.split(&mut rcc);
+    let dac1ch1 = dp.DAC1.constrain((gpioa.pa4, Dac1IntSig1), &mut rcc);
```

Comment:
Not sure how this should best be done.. There is no way for the user to know whether to specify Dac1IntSig1 or Dac1IntSig2(besides a not too nice compile error) in order to create a dac in with a mode compatible with the comparator.

Suggestions welcome :)